pytorch_lightning/utilities/fetching.py

```diff
@@ -308,26 +312,23 @@ def wait(self) -> None:
         event.wait()
 
 
-class StepFuncDataLoaderIter:
+class StepFuncDataLoaderIter(Iterator):
 
     """This class is a wrapper to keep track of dataloader iterator fetching event while left entirely to user
     control."""
 
-    def __init__(self, iterator: Iterator, data_fetcher: "AbstractDataFetcher"):
+    def __init__(self, iterator: Iterator, data_fetcher: "AbstractDataFetcher") -> None:
```

Comment:
Why can't we annotate the `data_fetcher` with the class directly? It's the same file, so shouldn't have cyclic import issues...